app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes, foreign_key: 'post_id'
+
+  def update_post_counter
+    author.update(posts_count: author.posts.count)
+  end
+
```

Comment:
[OPTIONAL] It is a good practice to run methods like this with rails callbacks like `:after_save` and make the method private.

Reply:
Thank you for the review @fobadara 
I will make sure to  implement the callbacks before merging.
Thank you 🙏🏻